openedx/core/djangoapps/enrollments/views.py

```diff
@@ -667,7 +668,11 @@ def get(self, request):
         filtered_data = []
         for enrollment in enrollment_data:
             course_key = CourseKey.from_string(enrollment["course_details"]["course_id"])
-            if user_has_role(request.user, CourseStaffRole(course_key)):
+            # TODO: remove roles check once course_roles is fully implemented and data is migrated
+            if (
+                user_has_role(request.user, CourseStaffRole(course_key)) or
+                request.user.has_perm(CourseRolesPermission.MANAGE_USERS_EXCEPT_ADMIN_AND_STAFF.perm_name, course_key)
+            ):
```

Comment:
Disclaimer: I have no prior experience, knowledge, or familiarity with this portion of the monolith and how the roles and permissions work.

Looking at it, I'm worried about possibly giving elevated permissions away? It looks like the `CourseStaffRole` is associated with the `staff` role. We're adding a conditional checking to see if the user has the `CourseStaff` role OR the `MANAGE_USERS_EXCEPT_ADMIN_AND_STAFF` permission. This, at surface level, appears to be giving additional permissions (potentially) to folks _outside_ the staff role. 

Is this expected or desired?

Reply:
@deborahgu I know you've spent some time looking at enrollments recently, any thoughts here?

Reply:
This code could potentially give access to someone outside the `CourseAccessRole` staff role if that new `CourseRoles` role was built with the `MANAGE_USERS_EXCEPT_ADMIN_AND_STAFF` permission, but that ability to create a new role and have it have the correct access without a major code change is one of the intentions of creating new roles based on permissions.

After the migrations, the only `CourseRoles` roles that will have `MANAGE_USERS_EXCEPT_ADMIN_AND_STAFF` are the `staff` and `limitedStaff` roles. Those are the two roles that currently have access with the `CourseStaffRole` check. The `CourseAccessRole` role of `LimitedStaff` has `staff` as a base role.